app/main.py

```diff
@@ -1,8 +1,22 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[self.name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people: list[dict]) -> list[Person]:
+    person_list = [Person(person["name"], person["age"]) for person in people]
+
+    for person in people:
+        if person.get("wife"):
+            Person.people[person.get("name")].wife \
+                = Person.people[person.get("wife")]
```

Comment:
also recommend use https://peps.python.org/pep-0572/#exceptional-cases

```suggestion
        if wife := person.get("wife"):
            Person.people[person.get("name")].wife = Person.people[wife]
```